### PR TITLE
detect/transform: from_base64 transform

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #10561 

This PR provides the `from_base64` transform used to match on content decoded with base64 using the mode (default rfc4648).

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6487](https://redmine.openinfosecfoundation.org/issues/6487)

Describe changes:
- New transform: `from_base64`
- Documentation
- Keyword parser in rust.

Updates:
- Address review comments, including adding the `SC` prefix for some items exported to C

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=pr/1670
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

File: rust/src/detect/transform_base64.rs [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+use nom7::bytes::complete::tag;
+use nom7::character::complete::multispace0;
+use nom7::sequence::preceded;
+use nom7::{Err, IResult};
+use std::str;
+
+#[repr(u8)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum DetectBase64Mode {
+    ModeRelax = 0,
```
Comment (catenacyber) [inline rust/src/detect/transform_base64.rs:34]: Could this replace the `Base64Mode` ? So that we do not confuse

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 19376

**Review by catenacyber — CHANGES_REQUESTED** [review]
Could you rebase to get green CI ? And see also my other comment...

**Comment by jlucovsky** [comment]
Continued in #10689

**Pull request closed without merging**